oauth2_http/java/com/google/auth/oauth2/GoogleCredentials.java

```diff
@@ -382,14 +391,51 @@ static Map<String, List<String>> addQuotaProjectIdToRequestMetadata(
     return Collections.unmodifiableMap(newRequestMetadata);
   }
 
+  protected void refreshTrustBoundaries(AccessToken newAccessToken) throws IOException {
+    if (!(this instanceof TrustBoundaryProvider)) {
+      return;
+    }
+    if (!TrustBoundary.isTrustBoundaryEnabled() || !isDefaultUniverseDomain()) {
+      return;
+    }
+
+    TrustBoundaryProvider provider = (TrustBoundaryProvider) this;
+    synchronized (lock) {
```

Comment:
I think we should avoid using a lock around a network call. Can we isolate the part that need sync/ locking for?